### PR TITLE
Some python 2 cleanup

### PR DESCRIPTION
**Pull request by lucas-wilkins**
Removed some of the python 2 legacy. Mostly superficial, added some type hints and renamed a few things.

I have changed some more fundamental things in the loaders, this is most likely place for bugs.

### CODE REVIEW

File: src/sas/sascalc/dataloader/loader.py
Comment (rprospero) [inline src/sas/sascalc/dataloader/loader.py:331]: I like the use of list comprehensions for these sorts of problems.

**Review by rprospero — CHANGES_REQUESTED** [review]
Mostly looks pretty good.  I've got a couple of stylistic comments, but I also wouldn't mind someone else who's worked on this part of the code taking a look before I merge this.

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py:105]: OK, probably a good thing I left it in. It should probably be handled in a better way in general, or at least there should be a comment that explains why its being done.

**Comment by lucas-wilkins** [comment]
@rprospero if there is someone you think should review it, can you ask for a review from them. I'll also flag this PR for the meeting today.

File: src/sas/sascalc/data_util/registry.py
Comment (rozyczko) [inline src/sas/sascalc/data_util/registry.py:113]: Won't this modify the ordering of the original list? It has just been sorted in line 106

File: src/sas/sascalc/data_util/registry.py
Comment (krzywon) [inline src/sas/sascalc/data_util/registry.py:81]: This is filtering out built-in readers from user-defined ones. It's possible for a user to associate a reader with another extension and user-defined associations should have no leading `.`. This is not currently exposed in the GUI.

File: src/sas/sascalc/data_util/registry.py
Comment (krzywon) [reply src/sas/sascalc/data_util/registry.py:113]: Good catch. Moving the sort to the return statement would fix this.

File: src/sas/sascalc/data_util/registry.py
Comment (krzywon) [inline src/sas/sascalc/data_util/registry.py:93]: Return type?

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py:113]: yeah, true.

File: src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py
Comment (lucas-wilkins) [reply src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py:657]: yeah, will sort that. Think it should be something like Union[some things that are checked] -> Iterable

File: src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py
Comment (lucas-wilkins) [reply src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py:664]: yeah

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py:93]: Can do, though I tend to follow the convention that return types need not be specified unless they differ from what would be inferred.

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py:69]: OK

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py]: Changed

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py]: Thanks

File: src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py
Comment (lucas-wilkins) [reply src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py:664]: done

File: src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py
Comment (lucas-wilkins) [reply src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py:657]: actually Any -> Union[list, np.ndarray] - done

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py:113]: fixed

File: src/sas/sascalc/data_util/registry.py
Comment (lucas-wilkins) [reply src/sas/sascalc/data_util/registry.py:66]: Lets discuss names tomorrow

**Review by rprospero — APPROVED** [review]

**Pull request merged**
